src/bt_stake_mint.js

```diff
@@ -215,6 +221,138 @@ class BTStakeMint {
     return messageHash;
   }
 
+  async requestStake(
+    originGatewayAddress,
+    stakeAmount,
+    staker,
+    beneficiary,
+    gasPrice,
+    gasLimit,
+  ) {
+    logger.info('Started requestStake');
+
+    const brandedToken = new ContractInteract.BrandedToken(
+      this.origin.web3,
+      this.chainConfig.brandedToken.address,
+    );
+    const mintBT = await brandedToken.convertToBrandedTokens(stakeAmount);
+
+    const stakerNonce = await new MosaicContractInteract.EIP20Gateway(
+      this.origin.web3,
+      originGatewayAddress,
+    ).getNonce(staker);
+
+    let stakeRequest = {
+      staker,
+      originGateway: originGatewayAddress,
+      beneficiary,
+      stakeVT: stakeAmount,
+      mintBT,
+      stakerNonce,
+      gasPrice,
+      gasLimit,
+    };
+
+    await brandedToken.requestStake(stakeAmount, { from: staker });
+
+    const stakeRequestHash = await brandedToken.contract.methods.stakeRequestHashes(
+      staker,
+    ).call();
+
+    const { stakeRequests } = this.chainConfig;
+
+    stakeRequest = {
+      stakeRequestHash,
+      ...stakeRequest,
+    };
+    stakeRequests[stakeRequestHash] = stakeRequest;
+
+    logger.info(`requestStake completed, your request hash is: ${stakeRequestHash}`);
+    return stakeRequestHash;
+  }
+
+  async acceptStake(stakeRequestHash) {
```

Comment:
Seems `acceptStake` and  `acceptStakeWithGatewayComposer` has potential duplicated code. A private method can be extracted for code sharing.

Reply:
Created `registerInternalActor` and `getAcceptStakeSignature` functions that has common code